src/App.jsx

```diff
@@ -1,51 +1,222 @@
+import './App.css';
+import './features/TodoList/TodoList';
+import './features/TodoForm';
+import TodoList from './features/TodoList/TodoList';
+import TodoForm from './features/TodoForm';
+import { useState, useEffect } from 'react';
 
-import './App.css'
-import './features/TodoList/TodoList'
-import './features/TodoForm'
-import TodoList from './features/TodoList/TodoList'
-import TodoForm from './features/TodoForm'
-import { useState } from 'react'
-
+const url = `https://api.airtable.com/v0/${import.meta.env.VITE_BASE_ID}/${import.meta.env.VITE_TABLE_NAME}`;
+const token = `Bearer ${import.meta.env.VITE_PAT}`;
 
 function App() {
-
+  const [isSaving, setIsSaving] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
   const [todoList, setTodoList] = useState([]);
-  // const [todos, setTodos] = useState([]);
 
-  function addTodo(title){
-    const newTodo =  { title, id: Date.now(), isCompleted: false};
-    setTodoList([...todoList, newTodo]);
-  }
+  useEffect(() => {
+    const fetchTodos = async () => {
+      setIsLoading(true);
+
+      const options = {
+        method: 'GET',
+        headers: {
+          Authorization: token,
+        },
+      };
 
-  function completeTodo (id){
-    const updateTodos = todoList.map((todo)=>{
-      if(todo.id === id){
-        return {...todo, isCompleted: true };
+      try {
+        const resp = await fetch(url, options);
+        if (!resp.ok) {
+          throw new Error(resp.statusText);
+        }
+        const { records } = await resp.json();
+
+        const todos = records.map((record) => {
+          const todo = {
+            id: record.id,
+            ...record.fields,
+          };
+          if (!todo.isCompleted) {
+            todo.isCompleted = false;
+          }
+          return todo;
+        });
+
+        setIsLoading(todos);
```

Comment:
Your fetch was totally correct, but your saved `todos` were not displaying on initial load, and it's because this line 46 actually needs to be `setTodoList(todos)`. You've already got `setIsLoading(true)` up on line 19. 

This lets you display the todos you’ve saved in Airtable. You successfully made your data persistent, now we need to show it off!